test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage2.Tests.ps1

```diff
@@ -0,0 +1,126 @@
+﻿using namespace System.Security.Cryptography.X509Certificates
+using namespace System.Security.Cryptography
+function New-CmsRecipient {
+  [CmdletBinding(SupportsShouldProcess = $true)]
+  param([String]$Name, [Switch]$Invalid)
+  $hash = [HashAlgorithmName]::SHA256
+  $pad = [RSASignaturePadding]::Pkcs1
+  $oids = [OidCollection]::new()
+  $oids.Add("1.3.6.1.4.1.311.80.1") | Out-Null
+  $ext1 = [X509KeyUsageExtension]::new([X509KeyUsageFlags]::DataEncipherment, $false)
+  $ext2 = [X509EnhancedKeyUsageExtension]::new($oids, $false)
+  $req = ([CertificateRequest]::new("CN=$Name", ([RSA]::Create(2048)), $hash, $pad))
+  if (!$Invalid) { ($ext1, $ext2).ForEach( { $req.CertificateExtensions.Add($_) }) }
+  return $req.CreateSelfSigned([datetime]::Now.AddDays(-1), [datetime]::Now.AddDays(365))
+}
+
+Describe "CmsMessage cmdlets using X509 cert" -Tags "CI" {
+
+  BeforeAll {
+    Write-Verbose  "Generating certs"
+    $vc1 = New-CmsRecipient "ValidCms1"
+    $vc2 = New-CmsRecipient "ValidCms2"
+    $ic = New-CmsRecipient "InvalidCms" -Invalid
+    $tmpfile = New-TemporaryFile
+  }
+
+  It " Encrypting with X509Cert" {
```

Comment:
Can we make sure we have tests for all the following variants of `-to`:

- An actual certificate (as retrieved from the certificate provider).
- Path to the file containing the certificate.
- Path to a directory containing the certificate.
- Thumbprint of the certificate (used to look in the certificate store).
- Subject name of the certificate (used to look in the certificate store).

Reply:
https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Security/Protect-CmsMessage?view=powershell-7

Reply:
Ok, added directory and base64 string tests. Other's were already there.

Reply:
Can you point me to the thumbprint tests?